stdlib/filecmp.pyi

```diff
@@ -14,7 +14,10 @@ BUFSIZE: Literal[8192]
 
 def cmp(f1: StrOrBytesPath, f2: StrOrBytesPath, shallow: int | bool = ...) -> bool: ...
 def cmpfiles(
-    a: AnyStr | PathLike[AnyStr], b: AnyStr | PathLike[AnyStr], common: Iterable[AnyStr], shallow: int | bool = ...
+    a: AnyStr | PathLike[AnyStr],
+    b: AnyStr | PathLike[AnyStr],
+    common: Iterable[AnyStr | PathLike[AnyStr]],
```

Comment:
We should disallow mixing string paths and bytes paths. For example, if `a` is a string and `common` contains bytes, it will end up doing `os.path.join(string, bytes)`, which errors.

This can be done with `@overload`. You would have one overload for string paths and another for bytes paths.

Reply:
Wouldn't adding an overload be redundant, because of how `AnyStr` is defined?

I may be mistaken, but if I understand type variables correctly, the current typing of `cmpfiles` should be equivalent to:

```python
@overload
def cmpfiles(a: bytes | PathLike[bytes], b: bytes | PathLike[bytes], common: Iterable[bytes | PathLike[bytes]], shallow: int | bool = ...) -> tuple[list[bytes], list[bytes], list[bytes]]: ...

@overload
def cmpfiles(a: str | PathLike[str], b: str | PathLike[str], common: Iterable[str | PathLike[str]], shallow: int | bool = ...) -> tuple[list[str], list[str], list[str]]: ...
```
```

Reply:
You are right :)